src/rules/anchor-ambiguous-text.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @fileoverview Enforce anchor text to not exactly match 'click here', 'here', 'link', or 'a link'.
+ * @author Matt Wang
+ * @flow
+ */
+
+// ----------------------------------------------------------------------------
+// Rule Definition
+// ----------------------------------------------------------------------------
+
+import type { JSXElement, Node } from 'ast-types-flow';
+import type { ESLintConfig, ESLintContext } from '../../flow/eslint';
+import { arraySchema, generateObjSchema } from '../util/schemas';
+import getElementType from '../util/getElementType';
+
+const AMBIGUOUS_WORDS = [
+  'click here',
+  'here',
+  'link',
+  'a link',
+];
+
+const errorMessage = `Ambiguous text within anchor. Screenreader users rely on link text for context; the words ${AMBIGUOUS_WORDS.join(',')} are ambiguous and do not provide enough context.`;
+
+const schema = generateObjSchema({
+  words: arraySchema,
+});
+
+const combineChildText = (node: JSXElement): string => node.children.reduce((str: string, child: Node) => {
+  switch (child.type) {
+    case 'Literal':
+      // $FlowFixMe child.value is too vague
+      return str + child.value;
+    // $FlowFixMe JSXText is missing in ast-types-flow
+    case 'JSXText':
+      return str + child.value;
+    default:
+      return str;
+  }
```

Comment:
```suggestion
  if (child.type === 'Literal' || child.type === 'JSXText) {
    return str + child.value;
  }
  return str;
```
switch statements should be avoided :-)